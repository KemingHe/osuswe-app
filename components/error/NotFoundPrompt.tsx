import { ArrowUpRightIcon } from '@heroicons/react/20/solid';
import Link from 'next/link';
import type { JSX } from 'react';

import {
  AUTH_SIGN_IN_ROUTE,
  USER_DASHBOARD_ROUTE,
} from '@/constants/routeConstants';

export default function NotFoundPrompt(): JSX.Element {
  return (
    <div className="flex flex-col justify-center items-center gap-4 m-4">
      <h1 className="flex flex-col text-center text-neutral font-extrabold">
        <span className="text-xl text-error">404</span>
        <span className="text-5xl">Page Not Found</span>
      </h1>
      <p className="text-center">Are you looking for one of these pages?</p>
      <div className="flex jusify-center items-center gap-4">
        <DashboardLinkButton />
        <SignInLinkButton />
      </div>
    </div>
  );
}

function DashboardLinkButton(): JSX.Element {
  return (
    <Link
      href={USER_DASHBOARD_ROUTE}
      className="btn btn-primary font-extrabold"
    >
      Dashboard
      <ArrowUpRightIcon className="size-5 -ms-2" aria-hidden="true" />
    </Link>
  );
}

function SignInLinkButton(): JSX.Element {
  return (
    <Link href={AUTH_SIGN_IN_ROUTE} className="btn btn-accent font-extrabold">
      Sign In
      <ArrowUpRightIcon className="size-5 -ms-2" aria-hidden="true" />
    </Link>
  );
}
