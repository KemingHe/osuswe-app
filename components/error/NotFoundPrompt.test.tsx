import { type RenderResult, render, screen } from '@testing-library/react';

import NotFoundPrompt from '@/components/error/NotFoundPrompt';

describe('NotFoundPrompt component', () => {
  it('renders a 404 error message and links to homepage and sign-in', () => {
    render(<NotFoundPrompt />);

    const notFoundCodeElement = screen.getByText('404');
    expect(notFoundCodeElement).toBeVisible();

    const notFoundMessageElement = screen.getByText('Page Not Found');
    expect(notFoundMessageElement).toBeVisible();

    const navigationPromptElement = screen.getByText(
      'Are you looking for one of these pages?',
    );
    expect(navigationPromptElement).toBeVisible();

    const buckeyeGPTButton = screen.getByRole('link', { name: 'Dashboard' });
    expect(buckeyeGPTButton).toBeVisible();

    const signInButton = screen.getByRole('link', { name: 'Sign In' });
    expect(signInButton).toBeVisible();
  });

  it('matches the snapshot', () => {
    const { asFragment }: RenderResult = render(<NotFoundPrompt />);
    expect(asFragment()).toMatchSnapshot();
  });
});
