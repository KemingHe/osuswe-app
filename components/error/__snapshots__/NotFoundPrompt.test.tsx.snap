// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFoundPrompt component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center gap-4 m-4"
  >
    <h1
      class="flex flex-col text-center font-extrabold"
    >
      <span
        class="text-xl text-error"
      >
        404
      </span>
      <span
        class="text-5xl"
      >
        Page Not Found
      </span>
    </h1>
    <p
      class="text-center"
    >
      Are you looking for one of these pages?
    </p>
    <div
      class="flex jusify-center items-center gap-4"
    >
      <a
        class="btn btn-primary font-extrabold"
        href="/user/dashboard"
      >
        Dashboard
        <svg
          aria-hidden="true"
          class="size-5 -ms-2"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M5.22 14.78a.75.75 0 0 0 1.06 0l7.22-7.22v5.69a.75.75 0 0 0 1.5 0v-7.5a.75.75 0 0 0-.75-.75h-7.5a.75.75 0 0 0 0 1.5h5.69l-7.22 7.22a.75.75 0 0 0 0 1.06Z"
            fill-rule="evenodd"
          />
        </svg>
      </a>
      <a
        class="btn btn-accent font-extrabold"
        href="/auth/sign-in"
      >
        Sign In
        <svg
          aria-hidden="true"
          class="size-5 -ms-2"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M5.22 14.78a.75.75 0 0 0 1.06 0l7.22-7.22v5.69a.75.75 0 0 0 1.5 0v-7.5a.75.75 0 0 0-.75-.75h-7.5a.75.75 0 0 0 0 1.5h5.69l-7.22 7.22a.75.75 0 0 0 0 1.06Z"
            fill-rule="evenodd"
          />
        </svg>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
